.github/workflows/dapr.yml

```diff
@@ -30,9 +30,15 @@ on:
       - release-*
       - feature/*
 jobs:
-  build:
-    name: "Build and test on ${{ matrix.job_name }}"
-    runs-on: "${{ matrix.os }}"
+  unit-tests:
```

Comment:
This is important to reduce flakiness. Mixing image building with unit tests creates situations where multiple runs start but they all get cancelled because unit test or lint failed in one of them. This stops this by making these validations upfront.

Reply:
Reverted back to running unit tests on all 3 OSes.